Registering and Documenting restartedAt annotation (#47064)
kubernetes/website: content/en/docs/reference/labels-annotations-taints/_index.md

to track changes. That mechanism has been superseded by
[Server-side apply](/docs/reference/using-api/server-side-apply/).

### kubectl.kubernetes.io/restartedAt {#kubectl-k8s-io-restart-at}

Type: Annotation

Example: kubectl.kubernetes.io/restartedAt: "2024-06-21T17:27:41Z"

> Used on: Pod

Reviewer: ```suggestion
Used on: Deployment, ReplicaSet, StatefulSet or DaemonSet, Pod
```

> This annotation contains the latest restart time of a resource(Deployment, ReplicaSet, StatefulSet or DaemonSet).

Reviewer: (nit)
```suggestion
This annotation contains the latest restart time of a resource (Deployment, ReplicaSet, StatefulSet or DaemonSet),
where kubectl triggered a rollout in order to force creation of new Pods.
```

The command `kubectl rollout restart <RESOURCE>` triggers a restart by patching the template
metadata of all the pods of resource with this annotation. You can also trigger a restart by manually

> setting or changing the value of this annotation. In above example the latest restart time of resource is shown as 21st Jun 2024 17:27:41 GMT.

Reviewer: optionally:
```suggestion
setting or changing the value of this annotation. In above example the latest restart time is shown as 21st June 2024 17:27:41 UTC.
<!-- localization note: feel free to represent the date / time here in a way that makes sense for your locale -->

You should not assume that this annotation represents the date / time of the most recent update;
a separate change could have been made since the last manually triggered rollout.

If you manually set this annotation on a Pod, nothing happens. The restarting side effect comes from
how workload management and Pod templating works.
```

to track changes. That mechanism has been superseded by
[Server-side apply](/docs/reference/using-api/server-side-apply/).

### kubectl.kubernetes.io/restartedAt {#kubectl-k8s-io-restart-at}

Type: Annotation

> Example: kubectl.kubernetes.io/restartedAt: "2024-06-21T17:27:41Z"

Reviewer 2: _Minor non-blocking nit_: Adjust the formatting of the example block to match the style of all the other examples on this [page](https://kubernetes.io/docs/reference/labels-annotations-taints/).

```suggestion
Example: `kubectl.kubernetes.io/restartedAt: "2024-06-21T17:27:41Z"`
```
